File: 1395-minimum-time-visiting-all-points/minimum-time-visiting-all-points.ts
function minTimeToVisitAllPoints(points: number[][]): number {
    let times = 0;

    for(let i = 1; i < points.length; i++) {
        const diffX = Math.abs(points[i][0] - points[i - 1][0])
        const diffY = Math.abs(points[i][1] - points[i - 1][1])
        times += Math.max(diffX, diffY);
    }
    return times;
};

File: 1395-minimum-time-visiting-all-points/minimum-time-visiting-all-points.rs
impl Solution {
    pub fn min_time_to_visit_all_points(points: Vec<Vec<i32>>) -> i32 {
        let mut time = 0;
        let mut prev = points.get(0).unwrap();
        for point in points.iter().skip(1) {
            let dx = (point[0] - prev[0]).abs();
            let dy = (point[1] - prev[1]).abs();
            time += dx.max(dy);
            prev = point; 
        }
        time
    }
}